File: Abigeapp.Domain/Fincas/Finca.cs
using Abigeapp.Domain.Usuarios;

namespace Abigeapp.Domain.Fincas;

public class Finca
{
    public Finca(string nombre, decimal latitud, decimal longitud)
    {
        Id = Guid.NewGuid();
        Nombre = nombre;
        Latitud = latitud;
        Longitud = longitud;
    }

    public Guid Id { get; set; }
    public string Nombre { get; set; }
    public decimal Latitud { get; set; }
    public decimal Longitud { get; set; }
    public ICollection<Usuario>? Usuarios { get; set; }
    public ICollection<Perimetro>? Perimetros { get; set; }
}

File: Abigeapp.Web/abigeapp.web.client/src/app/models/finca.ts
import { Alerta } from "./dispositivo";

export interface Finca {
    id: string;
    nombre: string;
    latitud: number;
    longitud: number;
    perimetros: Perimetro[];
}

export interface Perimetro {
    id: string;
    nommbre: string;
    tipo: string;
    coordenadas: Coordenada[];
}

export interface FincaConAlertas {
    alertas: Alerta[]
    total: number;
}

export interface Coordenada {
    orden: number;
    latitud: number;
    longitud: number;
}
